src/cardsavr/CardsavrHelper.ts

```diff
@@ -385,18 +385,30 @@ export class CardsavrHelper {
                         messages.body.map(async (item: any) => {
                             if (item.type === "job_status") {
                                 const handler = this._jobs.get(+item.job_id);
-                                if (handler) { handler(item); }
+                                if (handler) { 
+                                    handler(item); 
+                                    if (item.message.status.startsWith("PENDING_NEWCREDS") || item.message.status.startsWith("PENDING_TFA")) {
+                                        let tries = 2;
+                                        while (tries-- >= 0) {
+                                            const job = await session.getSingleSiteJobs(item.job_id, {}, {"x-cardsavr-hydration" : JSON.stringify(["credential_requests"]) });
+                                            if (job.body.credential_requests[0]) {
+                                                console.log("JOB IS PENDING " + item.message.status + " and there are " + job.body.credential_requests.length + " credential requests returned for this job");
+                                                handler(job.body.credential_requests[0]);
+                                                break;
+                                            } else if (tries == 1) {
+                                                console.log("JOB IS PENDING " + item.message.status + " and there are no credential requests, let's try one more time");
```

Comment:
In the case where after 2 tries we still don't get a credential request, can we return something that will indicate to the UI that the job is failed? Right now it will just hang

Reply:
It will time out after 4 minute and return "TIMEOUT_TFA" or "TIMEOUT_CREDENTIALS".

Reply:
I added an exception and it's definitely something you can catch.